lib/KTable/index.vue

```diff
@@ -0,0 +1,480 @@
+
+      const finalRows = computed(() => {
+        if (props.sortable) {
+          return useLocalSorting.value ? sortedRows.value : rows.value;
+        } else {
+          return rows.value;
+        }
+      });
+
+      watch(
+        () => props.rows,
+        newRows => {
+          rows.value = newRows;
+        }
+      );
+
+      const handleSort = index => {
+        if (headers.value[index].dataType === DATA_TYPE_OTHERS) {
+          return;
+        }
+        if (useLocalSorting.value) {
+          localHandleSort(index);
+        } else {
+          emit(
+            'changeSort',
+            index,
+            sortOrder.value === SORT_ORDER_ASC ? SORT_ORDER_DESC : SORT_ORDER_ASC
+          );
+        }
+      };
+
+      const getHeaderStyle = header => {
+        const style = {};
+        if (header.minWidth) style.minWidth = header.minWidth;
+        if (header.width) style.width = header.width;
+        return style;
+      };
+
+      return {
+        sortKey,
+        sortOrder,
+        finalRows,
+        handleSort,
+        getAriaSort,
+        SORT_ORDER_ASC,
+        SORT_ORDER_DESC,
+        DATA_TYPE_OTHERS,
+        getHeaderStyle,
+      };
+    },
+    /* eslint-disable kolibri/vue-no-unused-properties */
+    props: {
+      /**
+       * An array of objects representing the headers of the table. Each header should have a label and a dataType.
+       * The dataType can be one of 'string', 'numeric', 'date', or 'others'. MinWidth and width are optional.
+       */
+      headers: {
+        type: Array,
+        required: true,
+        validator: function(value) {
+          return value.every(
+            header =>
+              ['label', 'dataType'].every(key => key in header) &&
+              ['string', 'numeric', 'date', 'others'].includes(header.dataType)
+          );
+        },
+      },
+      /**
+       * An array of arrays representing the rows of the table. Each row should have the same number of elements as the headers array.
+       */
+      rows: {
+        type: Array,
+        required: true,
+      },
+      /**
+       * The caption of the table
+       */
+      caption: {
+        type: String,
+        required: true,
+      },
+      /**
+       * Enables or disables local sorting of the table data.
+       */
+      useLocalSorting: {
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * Enables or disables sorting functionality for the table headers.
+       */
+      sortable: {
+        type: Boolean,
+        default: true,
+      },
+    },
+    data() {
+      return {
+        focusedRowIndex: null,
+        focusedColIndex: null,
+        hoveredRowIndex: null,
+      };
+    },
+    computed: {
+      coreOutlineFocus() {
+        return {
+          ':focus': {
+            ...this.$coreOutline,
+            outlineOffset: '-2px',
+          },
+        };
+      },
+    },
+    methods: {
+      handleKeydown(event, rowIndex, colIndex) {
```

Comment:
Related to the comment about summary docstrings for functions, to provide one more example I think here we will benefit from a high-level info. Something like:

```suggestion
/**
* Takes care of
*  - keyboard navigation focus trap
*  - the arrow keys navigation flow
*  - the tab keys navigation flow
*  - access to focusable elements within a cell via tab and shift tab keys
* - triggering sort on the enter key
*  - header highlight
*/
      handleKeydown(event, rowIndex, colIndex) {
```